### PR TITLE
added ?externalFileUrl

### PR DESCRIPTION
**Pull request by kahneraja**

### CODE REVIEW

**Comment by kahneraja** [comment]
#46

**Comment by kahneraja** [comment]
works locally with..
[http://localhost:8080/?externalFileUrl=https%3A%2F%2Fdoc-0o-3k-docs.googleusercontent.com%2Fdocs%2Fsecuresc%2Fha0ro937gcuc7l7deffksulhg5h7mbp1%2Fbbjf3pdb50abkcv9198dp4fdlkknflcf%2F1470967200000%2F05178397473083625167%2F*%2F0B2_ZE7eTmet4Ym1GNWQzaVlyME0%3Fe%3Ddownload](http://localhost:8080/?externalFileUrl=https%3A%2F%2Fdoc-0o-3k-docs.googleusercontent.com%2Fdocs%2Fsecuresc%2Fha0ro937gcuc7l7deffksulhg5h7mbp1%2Fbbjf3pdb50abkcv9198dp4fdlkknflcf%2F1470967200000%2F05178397473083625167%2F*%2F0B2_ZE7eTmet4Ym1GNWQzaVlyME0%3Fe%3Ddownload)

File: src/components/MainView.js [hunk 1]
```diff
@@ -24,16 +24,37 @@ export default class MainView extends React.Component {
...
+
+  import() {
+
+    const localFolder = './csv/'
+    const localFileName = this.getLocalFileName();
+    let path = localFolder + localFileName;
+
+    const externalFileUrl = this.getExternalFileUrl();
+    if (externalFileUrl)
+      path = externalFileUrl;
```
Comment (alexyaseen) [inline src/components/MainView.js:60]: If I recall correctly, stylistically we decided we should always use {} for if statements

**Comment by alexyaseen** [comment]
It works for me when the URL is valid, but if invalid it just doesn't load anything. Looks like the error case isn't being handled yet. Otherwise works well!

**Comment by kahneraja** [comment]
image

**Comment by kahneraja** [comment]
Should be all sorted now :)

File: src/components/MainView.js [hunk 2]
```diff
@@ -78,13 +110,22 @@ export default class MainView extends React.Component {
...
+    const data = this.state.data,
+      filteredData = this.state.filteredData,
+      dataSource = encodeURI(this.state.dataSource),
+      isError = this.state.isError
+    
     return (
       <div>
         <Navbar />
     		<div className="container">
+          <div className={"row " + (isError ? '' : 'hidden')}>
```
Comment (samuelcole) [inline src/components/MainView.js:122]: Maybe don't even render this if there is no error?

**Comment by samuelcole** [comment]
+1 to the code!

**Comment by decause** [comment]
:+1: merge.

**Pull request merged**